libfwupdplugin/fu-device.c

```diff
@@ -1360,18 +1360,41 @@ fu_device_set_id (FuDevice *self, const gchar *id)
 	}
 }
 
+/**
+ * fu_device_set_version_format:
+ * @self: A #FwupdDevice
+ * @fmt: the #FwupdVersionFormat, e.g. %FWUPD_VERSION_FORMAT_PLAIN
+ *
+ * Sets the version format.
+ *
+ * Since: 1.3.9
+ **/
+void
+fu_device_set_version_format (FuDevice *self, FwupdVersionFormat fmt)
+{
+	/* same */
+	if (fu_device_get_version_format (self) == fmt)
+		return;
+	if (fu_device_get_version_format (self) != FWUPD_VERSION_FORMAT_UNKNOWN) {
+		g_debug ("changing verfmt for %s: %s->%s",
+			 fu_device_get_id (self),
+			 fwupd_version_format_to_string (fu_device_get_version_format (self)),
+			 fwupd_version_format_to_string (fmt));
+	}
+	fwupd_device_set_version_format (FWUPD_DEVICE (self), fmt);
+}
+
 /**
  * fu_device_set_version:
  * @self: A #FuDevice
  * @version: (allow-none): a string, e.g. `1.2.3`
- * @fmt: a #FwupdVersionFormat, e.g. %FWUPD_VERSION_FORMAT_TRIPLET
  *
  * Sets the device version, sanitizing the string if required.
  *
  * Since: 1.2.9
  **/
 void
-fu_device_set_version (FuDevice *self, const gchar *version, FwupdVersionFormat fmt)
+fu_device_set_version (FuDevice *self, const gchar *version)
```

Comment:
What do you think about making this return true/false?  Now we have an order of operations that version format must be set before versions, so I think it would be good to not hide silent errors in the logs.

Reply:
if it returns with an error, it also needs a GError -- but this isn't anything the daemon or plugin can handle in a sane way. This is the kind of thing you can only really catch when developing the plugin.